src/Hintman/HLint.hs

```diff
@@ -89,3 +92,30 @@ createFileDownloadUrl PrInfo{..} (toText -> file) = T.intercalate "/"
     , unBranch prInfoBranch
     , file
     ]
+
+{- |
+
+TODO: multiline hints support
+-}
+createComment :: Idea -> Maybe Text
+createComment Idea{..} = ideaTo >>= \to ->
+    if ideaSeverity == Ignore
+    then Nothing
+    else Just $ unlines $
+        (if ideaHint == "" then "" else show ideaSeverity <> ": " <> toText ideaHint)
+        : mkSuggestion to
+       ++ [ "Note: " <> n | let n = showNotes ideaNote, n /= ""]
+  where
+    mkSuggestion :: String -> [Text]
+    mkSuggestion to =
+        [ "```suggestion"
+        , toText to
+        , "```"
+        ]
+
+    showNotes :: [Note] -> Text
+    showNotes = T.intercalate ", " . map show . filter use
+      where
+        use :: Note -> Bool
+        use ValidInstance{} = False -- Not important enough to tell an end user
```

Comment:
What is valid instance?

Reply:
I'm not sure. I copy pasted this function from HLint sources as it is not exported. Probably should add Note about that.